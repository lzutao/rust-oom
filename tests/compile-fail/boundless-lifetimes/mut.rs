use oom::NonEmptySlice;
fn main() {
    let mut foo = String::from("hello");
    let s = NonEmptySlice::from_mut(&mut foo);
    let bad: &'static [_] = s.as_mut_slice();
    drop(foo);
    let _dangling = bad.first();
}
