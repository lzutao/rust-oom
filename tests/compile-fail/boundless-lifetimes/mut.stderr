error[E0597]: `foo` does not live long enough
 --> $DIR/mut.rs:4:37
  |
4 |     let s = NonEmptySlice::from_mut(&mut foo);
  |                                     ^^^^^^^^ borrowed value does not live long enough
5 |     let bad: &'static [_] = s.as_mut_slice();
  |              ------------ type annotation requires that `foo` is borrowed for `'static`
...
8 | }
  | - `foo` dropped here while still borrowed

error[E0505]: cannot move out of `foo` because it is borrowed
 --> $DIR/mut.rs:6:10
  |
4 |     let s = NonEmptySlice::from_mut(&mut foo);
  |                                     -------- borrow of `foo` occurs here
5 |     let bad: &'static [_] = s.as_mut_slice();
  |              ------------ type annotation requires that `foo` is borrowed for `'static`
6 |     drop(foo);
  |          ^^^ move out of `foo` occurs here
